prov/gni/test/rdm_sr.c

```diff
@@ -388,23 +389,22 @@ void do_sendmsg(int len)
 	sz = fi_recv(ep[1], target, len, rem_mr, gni_addr[0], source);
 	cr_assert_eq(sz, 0);
 
-	while ((ret = fi_cq_read(msg_cq[0], &cqe, 1)) == -FI_EAGAIN) {
-		pthread_yield();
-	}
-
-	cr_assert_eq(ret, 1);
-	cr_assert_eq((uint64_t)cqe.op_context, (uint64_t)target);
-
-	dbg_printf("got send context event!\n");
-
-	while ((ret = fi_cq_read(msg_cq[1], &cqe, 1)) == -FI_EAGAIN) {
-		pthread_yield();
-	}
+	/* need to progress both CQs simultaneously for rendezvous */
```

Comment:
This same code has been duplicated in several sections. Perhaps it should be moved into a static inline function.